### PR TITLE
port: fix failing test + support extended token

### PR DESCRIPTION
**Pull request by beesaferoot**

### CODE REVIEW

File: src/token.js [hunk 1]
```diff
@@ -70,32 +70,104 @@ class OpenPAYGOTokenShared {
...
-        // 4 - digit token value
-        let bitArray = []
-        for (let digit of token.toString()) {
-            digit = Number(digit) - 1
-            const tempArr = bitArrayFromInt(digit, 2)
-            bitArray = bitArray.concat(tempArr)
-        }
-        return bitArrayToInt(bitArray)
+        return convertFrom4digitToken(token)
     }
```
Comment (dmohns) [inline src/token.js:74]: Is it required to have the function `convertFrom4digitToken` twice? As a class function and as a global? This functional declaration seems a bit odd to the viewer.

Comment (beesaferoot) [reply src/token.js:74]: It does look odd, but the reason for it is to avoid duplicating the function in both classes (shared/extended)

Comment (dmohns) [reply src/token.js:74]: Ok, understood 🎉

**Review by dmohns — APPROVED** [review]
Thanks a lot!

**Pull request merged**